### PR TITLE
Add some documentation for ProtoBufSchemaGenerator in formats.md

### PR DESCRIPTION
**Pull request by justinhj**
Hi, this is a PR for #1778

I've added a bit of documentation in `formats.md` about the `ProtoBufSchemaGenerator`. Just enough to help the interested party see that it exists, an example and a link to the code. 

If you think it will be useful I'll be happy to make any requested changes.

It is targeted at master since it is documentation only, but it looks like it will cleanly merge with dev if you prefer.

BTW Knit fails with the following, I'm not sure why.

```
> Task :knit FAILED
ERROR: /Users/justin.heyes-jones/projects/kotlinx.serialization/docs/formats.md: 486: Previous test was not emitted with TEST
```

### CODE REVIEW

**Review by sandwwraith — CHANGES_REQUESTED** [review]

File: docs/formats.md [hunk 1]
```diff
@@ -428,6 +428,47 @@ Field #1: 08 Varint Value = 2, Hex = 02
 Field #1: 08 Varint Value = 3, Hex = 03
 ```
 
+### ProtoBuf schema generator (experimental)
+
+As mentioned above, when working with protocol buffers you usually use a ".proto" file and use a code generator for your 
+language includes the code to your message to an output stream and deserialize it from an input stream.  When using
```
[the remark this answers is not in the document]
Comment (justinhj) [reply docs/formats.md]: Thanks I'm sure I proof read it multiple times but then committed total nonsense.

**Comment by sandwwraith** [comment]
Have you re-run `knit`? It looks like the commit should also have generated example files, since their numeration got updated

**Comment by sandwwraith** [comment]
Your example should be separate compilable unit:

```
e: /opt/buildAgent/work/b2fef8360e1bcf3d/guide/example/example-formats-08.kt: (13, 1): Expecting a top level declaration
e: /opt/buildAgent/work/b2fef8360e1bcf3d/guide/example/example-formats-08.kt: (13, 8): Expecting a top level declaration
e: /opt/buildAgent/work/b2fef8360e1bcf3d/guide/example/example-formats-08.kt: (13, 9): Expecting a top level declaration
e: /opt/buildAgent/work/b2fef8360e1bcf3d/guide/example/example-formats-08.kt: (13, 16): Expecting a top level declaration
e: /opt/buildAgent/work/b2fef8360e1bcf3d/guide/example/example-formats-08.kt: (4, 2): Cannot access 'Serializable': it is internal in 'kotlin.io'
e: /opt/buildAgent/work/b2fef8360e1bcf3d/guide/example/example-formats-08.kt: (4, 2): This class does not have a constructor
e: /opt/buildAgent/work/b2fef8360e1bcf3d/guide/example/example-formats-08.kt: (11, 37): Unresolved reference: serializer
e: /opt/buildAgent/work/b2fef8360e1bcf3d/guide/example/example-formats-08.kt: (12, 15): Unresolved reference: ProtoBufSchemaGenerator
e: /opt/buildAgent/work/b2fef8360e1bcf3d/guide/test/FormatsTest.kt: (64, 70): Unresolved reference: main
```

I suggest adding `import kotlinx.serialization.*` and other required imports and wrapping `vals`/`println` into `fun main()`. Then update Knit and run `./gradlew :guide:test knitCheck` to check that everything is ok

**Comment by justinhj** [comment]
> ./gradlew :guide:test knitCheck

Thanks for your patience! I wasn't familiar with what knit did. Have now fixed the knit related issues.

**Review by sandwwraith — APPROVED** [review]
Thanks for all your efforts and time!

**Pull request merged**